src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T [] arr;
+    private int size;
+    private int currentCapacity = DEFAULT_CAPACITY;
+
+    public ArrayList() {
+        arr = (T[]) new Object [DEFAULT_CAPACITY];
+    }
+
+    public void increaseCapacity() {
```

Comment:
Private ones (methods, classes) should be after public ones.
```suggestion
    private void increaseCapacity() {
```